coreapps-web-2.2/src/test/java/org/openmrs/module/coreapps/diagnosis/DiagnosisServiceCompatibilityTest.java

```diff
@@ -0,0 +1,52 @@
+package org.openmrs.module.coreapps.diagnosis;
+
+
+
+import java.util.Date;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+
+public class DiagnosisServiceCompatibilityTest {
+	
+	
+	@Test
+	public void getPrimaryDiagnosis_shouldReturnTrueOnAnExistingDiagnosis() throws Exception {
+	
+    Encounter encounter = new Encounter();
+	Date date = new Date();
+	encounter.setEncounterDatetime(date);
+	
+	Location location = new Location(1);
+	encounter.setLocation(location);
+	
+	Patient patient = new Patient(1);
+	encounter.setPatient(patient);
+	
+	Diagnosis diagnosis = new Diagnosis();
+	diagnosis.setDiagnosisId(2);
+	
+	Set<Diagnosis> diag = new HashSet<Diagnosis>();
+	diag.add(diagnosis);
+	
+	encounter.setDiagnoses(diag);
+	
+	DiagnosisServiceCompatibility2_2 diagnosisServiceCompatibility = new DiagnosisServiceCompatibility2_2();
+	
+	List<org.openmrs.module.emrapi.diagnosis.Diagnosis> diagnoses = diagnosisServiceCompatibility.getPrimaryDiagnosis(encounter);
+	 
+	//make sure the daignosis exists
+	Assert.assertEquals(1,diagnoses.size());
```

Comment:
while addressing @ibacher's comments on this pr,for starters i want to make sure that any method that i use to get diagnosis does not return a null in my sub module,i went further to first write a test for .getPrimaryDiagnosis(encounter),in my submodule,this returns org.openmrs.api.APIException: Service not found: interface org.openmrs.api.DiagnosisService  full logs https://pastebin.com/y6J0UEaa  I have read some threads on fixing this some which were responded to by @dkayiwa ,unfortutanetly these have not worked for me.@dkayiwa @ibacher  do you think am missing out something probably in my pom of the submodule?,thanks

Reply:
> while addressing @ibacher's comments on this pr,for starters i want to make sure that any method that i use to get diagnosis does not return a null in my sub module,i went further to first write a test for .getPrimaryDiagnosis(encounter),in my submodule,this returns org.openmrs.api.APIException: Service not found: interface org.openmrs.api.DiagnosisService full logs https://pastebin.com/y6J0UEaa I have read some threads on fixing this some which were responded to by @dkayiwa ,unfortutanetly these have not worked for me.@dkayiwa @ibacher do you think am missing out something probably in my pom of the submodule?,thanks

problem was with my dependencies which i have fixed in my pom